### DISCUSSION TITLE
Push message to IOTA Tangle

### DISCUSSION THREAD

**Discussion by khiem20tc** [iotaledger/identity.rs#1402] [Q&A]
Dear team,

After gen DID sucessful and CID from IPFS, I would like to push as message to IOTA Tangle. And I followed this code `https://github.com/iotaledger/iota.js/blob/dev/packages/iota/examples/data/src/index.ts` and also using `https://api.testnet.iotaledger.net` RPC API Node.

```

const {
  SingleNodeClient,
  sendData,
  retrieveData,
  messagesFind,
} = require("@iota/iota.js");
const { Converter } = require("@iota/util.js");

async function sendMessage(payload) {
  const client = new SingleNodeClient(API_ENDPOINT, {
    headers: {
      Authorization: `Bearer ${JWT_TOKEN}`,
    },
  });

  // Define the index and convert it to bytes
  const myIndex = Converter.utf8ToBytes("CID-DID");

  const sendResult = await sendData(
    client,
    myIndex,
    Converter.utf8ToBytes(payload)
  );
  console.log("Received Message Id", sendResult.messageId);
  console.log(
    `    https://explorer.iota.org/iota-testnet/message/${sendResult.messageId}`
  );

  return sendResult.messageId;
}
```

But I recived the erorr that 
`Error in main function ClientError: invalid or expired jwt, error: code=401, message=invalid or expired jwt, internal=signature is invalid`

Can anyone help me with pushing messages or data to the IOTA Tangle?

**Comment by eike-hass** [comment]
You should not need to pass any authorization for the public nodes. As this discussion is unrelated to this repository I will close it here. Feel free to re-open in https://github.com/iotaledger/iota.js
